vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -164,10 +169,18 @@ def _get_accepted(
         selected_target_probs = target_probs[batch_indices, probs_indicies,
                                              draft_token_ids]
 
-        uniform_rand = torch.rand(batch_size,
-                                  k,
-                                  dtype=self.probs_dtype,
-                                  device=target_probs.device)
+
+        if any(generator is not None for generator in generators):
+            uniform_rand = torch.empty(batch_size, k, dtype=self.probs_dtype, device=target_probs.device)
+            for i, generator in enumerate(generators):
+                uniform_rand[i,:] = torch.rand(1, k, dtype=self.probs_dtype, device=target_probs.device, generator=generator)
```

Comment:
I think it might be preferable here in the mixed case to do a single rand for the non-seed requests, since it could be likely that most requests don't have a seed. This can be done via something like
```python
uniform_rand[non_seed_indices] = torch.rand(batch_size, len(non_seed_indices), ...
```

Reply:
I agree. Sort-of-related question: do you know whether we can safely assume that the length of each SequenceGroup is exactly 1 here? In the general sampler code there is [some logic](https://github.com/vllm-project/vllm/blob/4050d646e5221a516c93384b047e10b22d7167e7/vllm/model_executor/layers/sampler.py#L443-L448) to handle the more general case. Maybe we need that here too?

Reply:
Will double check, I think it's only >1 for beam search which I think is explicitly unsupported with spec decode right now anyhow. Though it might also be the case for parallel decoding (`n`>1 i.e. multiple output seqs per single input seq).

Reply:
I actually just tested that case: combining `n>1` with spec. decode and it seems to fail on main:
```
    |   File "/home/zrltpa/vllm/vllm/model_executor/layers/sampler.py", line 96, in forward
    |     sample_results, maybe_sampled_tokens_tensor = _sample(
    |                                                   ^^^^^^^^
    |   File "/home/zrltpa/vllm/vllm/model_executor/layers/sampler.py", line 658, in _sample
    |     return _sample_with_torch(
    |            ^^^^^^^^^^^^^^^^^^^
    |   File "/home/zrltpa/vllm/vllm/model_executor/layers/sampler.py", line 528, in _sample_with_torch
    |     sampled_token_ids_tensor[
    | RuntimeError: shape mismatch: value tensor of shape [2] cannot be broadcast to indexing result of shape [1, 1]
```
If that case isn't supported by spec decode currently we should at least give a reasonable message back.

Reply:
made the suggested change